direct/src/showbase/DirectObject.py

```diff
@@ -94,12 +94,14 @@ def detectLeaks(self):
         if hasattr(self, '_taskList'):
             tasks = [task.name for task in self._taskList.values()]
         if len(events) or len(tasks):
-            estr = choice(len(events), 'listening to events: %s' % events, '')
-            andStr = choice(len(events) and len(tasks), ' and ', '')
-            tstr = choice(len(tasks), '%srunning tasks: %s' % (andStr, tasks), '')
+            estr = ('listening to events: %s' % events if len(events) else '')
+            andStr = (' and ' if len(events) and len(tasks) else '')
+            tstr = ('%srunning tasks: %s' % (andStr, tasks) if len(tasks) else '')
             notify = directNotify.newCategory('LeakDetect')
-            func = choice(getRepository()._crashOnProactiveLeakDetect,
-                          self.notify.error, self.notify.warning)
+            crash = False
+            if hasattr(getRepository(), '_crashOnProactiveLeakDetect'):
+                crash = getRepository()._crashOnProactiveLeakDetect
```

Comment:
You can just replace these three lines with `crash = getattr(getRepository(), '_crashOnProactiveLeakDetect', False)`

Reply:
Thanks for the hint. Changed it accordingly.